source/features/deep-blame.tsx

```diff
@@ -0,0 +1,140 @@
+import './deep-blame.css';
+import blurAccessibly from 'blur-accessibly';
+import cache from 'webext-storage-cache';
+import delegate, {DelegateEvent} from 'delegate-it';
+import React from 'dom-chef';
+import versionIcon from 'octicon/versions.svg';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoGQL, getReference} from '../libs/utils';
+
+const getPullRequestBlameCommit = cache.function(async (commit: string, prNumber: number): Promise<string | false> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			object(expression: "${commit}") {
+				... on Commit {
+					associatedPullRequests(last: 1) {
+						nodes {
+							number
+							mergeCommit {
+								oid
+							}
+							commits(last: 1) {
+								nodes {
+									commit {
+										oid
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const {nodes} = repository.object.associatedPullRequests;
+	const mergeCommit = nodes[0].mergeCommit.oid;
+	const pullRequestNumber = nodes[0].number;
+
+	if (pullRequestNumber === prNumber && mergeCommit === commit) {
+		return nodes[0].commits.nodes[0].commit.oid;
+	}
+
+	return false;
+}, {
+	cacheKey: ([commit]) => __featureName__ + commit
+});
+
+function getCommitHash(commit: HTMLElement): string {
```

Comment:
```suggestion
export function getCommitHash(commit: HTMLElement): string {
```

Lets reuse this function in other features. No need to move to `utils` because it's so specific

Reply:
The function is used differently. Its taking the next element. By the tags and the merge commits we took it from the dataset .

https://github.com/sindresorhus/refined-github/blob/4f25a67d0688afe55c85f95f1ca720c172553150/source/features/mark-merge-commits-in-list.tsx#L33-L35

Reply:
✅